iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergStorageHandler.java

```diff
@@ -1840,7 +1841,8 @@ private List<String> getPartitions(DDLOperationContext context, Configuration jo
       fetcher.initialize(job, HiveTableUtil.getSerializationProps());
       org.apache.hadoop.hive.ql.metadata.Table metaDataPartTable =
           context.getDb().getTable(hmstbl.getDbName(), hmstbl.getTableName(), "partitions", true);
-      Deserializer currSerDe = metaDataPartTable.getDeserializer();
+      Deserializer currSerDe = HiveMetaStoreUtils.getDeserializer(job, metaDataPartTable.getTTable(),
+          metaDataPartTable.getMetaTable(), false);
```

Comment:
why do we need metatable?

Reply:
iceberg `show partitons` will use the iceberg metadata table `partitions`.